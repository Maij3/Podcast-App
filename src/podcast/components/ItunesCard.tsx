import { Grid, Box, Typography } from "@mui/material";
import { FC } from "react";
import { Link } from "react-router-dom";

interface Props {
  artist: string;
  url_img: string;
  autor: string;
  id: string;
}

export const ItunesCard: FC<Props> = ({ artist, url_img, autor, id }) => {
  return (
    <Grid key={artist} item xs={12} sm={4} lg={3}>
      <Link to={`/podcast/${id}`} style={{ width: "100%" , textDecoration:"none" , color: "unset"}}>
        <Box
          position={`relative`}
          margin={"auto"}
          marginBottom={`15vh`}
          width={`100%`}
        >
          <img
            src={url_img}
            alt={url_img}
            width={100}
            className="itunes-list-img"
            height={`130px`}
            style={{
              margin: "auto",
              display: "block",
              borderRadius: "100%",
              width: "50%",
              position: "absolute",
              top: "0%",
              left: "50%",
              transform: "translate(-50%, -50%)",
            }}
          />
          <Box
            sx={{
              display: "flex",
              justifyContent: "center",
              flexDirection: "column",
              alignItems: "center",
              textAlign: "center",
              marginTop: "10px",
              boxShadow: "0px 2px 10px 2px #cecece",
              padding: "80px  10px 20px 10px",
            }}
          >
            <Typography variant="h6">{artist}</Typography>
            <Typography>{`Author: ${autor}`}</Typography>
          </Box>
        </Box>
      </Link>
    </Grid>
  );
};
