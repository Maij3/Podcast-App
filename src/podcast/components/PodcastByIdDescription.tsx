import { Card, CardContent, CardMedia, Typography } from "@mui/material";
import { FC } from "react";
import { generateLink } from "../../helpers";

interface Props {
  audio: string;
  description: string;
  name: string;
}

export const PodcastByIdDescription: FC<Props> = ({
  audio,
  description,
  name,
}) => {
  description = generateLink(description);
  return (
    <Card sx={{ marginBottom: "20px" }}>
      <CardContent>
        <Typography variant="h4">{name}</Typography>
        <p dangerouslySetInnerHTML={{ __html: description }}></p>
        <audio src={audio} controls style={{ width: "100%" }}></audio>
      </CardContent>
    </Card>
  );
};
