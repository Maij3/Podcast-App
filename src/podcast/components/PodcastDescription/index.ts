export * from "./PodcastSideDescription";
export * from "./PodcastTableDescription";