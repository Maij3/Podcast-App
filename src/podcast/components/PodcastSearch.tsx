import React from 'react'

export const PodcastSearch = () => {
  return (
    <div>PodcastSearch</div>
  )
}
