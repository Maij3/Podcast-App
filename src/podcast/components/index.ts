export * from "./ItunesCard";
export * from "./PodcastSearch";
export * from "./PodcastDetails";
export * from "./PodcastById";