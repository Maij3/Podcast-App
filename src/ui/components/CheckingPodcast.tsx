import { CircularProgress, Grid } from "@mui/material";

export const CheckingAuth = () => {
  return (
    <Grid
      container
      spacing={0}
      direction="column"
      alignItems="center"
      justifyContent="center"
      sx={{ minHeight: "100vh", backgroundColor: "white", padding: 4 }}
    >
      <Grid item  >
        <CircularProgress color='warning' />
      </Grid>
    </Grid>
  );
};