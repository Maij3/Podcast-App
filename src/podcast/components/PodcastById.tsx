import React from "react";
import { useEffect } from "react";
import { useParams } from "react-router-dom";
import { useGetByIdPodcastQuery } from "../../store/apis";
import useActions from "../../hooks/useActions";
import { useTypedSelector } from "../../hooks/useTypedSelector";
import { RootState } from "../../store";
import { filterPodcastById } from "../../helpers/filterPodcastById";
import { Box, Grid } from "@mui/material";
import { PodcastSideDescription } from "./PodcastDescription";
import { PodcastByIdDescription } from "./PodcastByIdDescription";

export const PodcastById = () => {
  const { id, episodeId } = useParams();
  const { data, isSuccess } = useGetByIdPodcastQuery(id);
  useEffect(() => {
    setpodcastById(isSuccess && data.results);
  }, [data, isSuccess]);
  const { setpodcastById } = useActions();

  const podcastById = useTypedSelector(
    (state: RootState) => state.podcast.podcastById
  );
  const filterById = filterPodcastById(podcastById, episodeId);
  console.log(filterById);
  return (
    <Box marginTop={"40px"}>
      <Grid container spacing={2} justifyContent={"space-between"}>
        {filterById.map((item: any, key: any) => {
          const img = item.artworkUrl600;
          const audio = item.episodeUrl;
          const description = item.description;
          const name = item.trackName;
          return (
            <React.Fragment key={`${key}-${name}`}>
              <Grid item xs={12} md={3} >
                <PodcastSideDescription
                  img={img}
                  description={description}
                  name={name}
                />
              </Grid>
              <Grid item xs={12} md={8}>
                <PodcastByIdDescription
                  audio={audio}
                  description={description}
                  name={name}
                />
              </Grid>
            </React.Fragment>
          );
        })}
      </Grid>
    </Box>
  );
};
