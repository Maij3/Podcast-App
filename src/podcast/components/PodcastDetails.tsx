import React from "react";
import { Box, Card, CardContent, Grid, Typography } from "@mui/material";
import { PodcastSideDescription } from "./PodcastDescription";
import { ThemeTable } from "../../ui";
import { useParams } from "react-router-dom";
import { useGetByIdPodcastQuery } from "../../store/apis";
import { CheckingAuth } from "../../ui/components/CheckingPodcast";

export const PodcastDetails = () => {
  const { id } = useParams();
  const { data, isSuccess } = useGetByIdPodcastQuery(id);
  console.log({data})
  return (
    <Box marginTop={"40px"}>
      <Grid container spacing={2} justifyContent={"space-between"}>
        {isSuccess ? (
          data.results.map((item: any, key: any) => {
            const img = data.results[0].artworkUrl600;
            const description = data.results[key].description;
            const name = data.results[0].artistName;
            {
              if (key === 1) {
                return (
                  <React.Fragment key={key}>
                    <Grid item xs={3}>
                      <PodcastSideDescription
                        img={img}
                        description={description}
                        name={name}
                      />
                    </Grid>
                    <Grid item xs={8}>
                      <Card sx={{ marginBottom: "20px" }}>
                        <CardContent>
                          <Typography fontWeight={"bold"} variant="h4">
                            {isSuccess && `Episodies:${data.resultCount}`}
                          </Typography>
                        </CardContent>
                      </Card>
                      <ThemeTable results={isSuccess && data.results} />
                    </Grid>
                  </React.Fragment>
                );
              }
            }
          })
        ) : (
          <CheckingAuth />
        )}
      </Grid>
    </Box>
  );
};
