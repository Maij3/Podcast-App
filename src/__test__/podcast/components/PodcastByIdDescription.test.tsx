import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { PodcastByIdDescription } from "../../../podcast/components/PodcastByIdDescription";

describe("<PodcastByIdDescription/>", () => {
  //Snapshot Itunes Card
  it("PodcastByIdDescription snapshot ", () => {
    const { container } = render(
      <MemoryRouter>
        <PodcastByIdDescription
          audio={
            "https://www.podtrac.com/pts/redirect.mp3/chrt.fm/track/35F11F/traffic.megaphone.fm/MOWM5486612603.mp3?updated=1684229072"
          }
          description={
            "It's Episode 50 of the Friday Night Karaoke Podcast, and the theme of the week was #FNKDedication."
          }
          name={"Friday Night Karaoke"}
        />
      </MemoryRouter>
    );
    screen.debug();
    expect(container).toMatchSnapshot();
  });
});
