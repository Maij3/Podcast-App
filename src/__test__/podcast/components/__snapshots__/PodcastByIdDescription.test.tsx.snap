// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PodcastByIdDescription/> PodcastByIdDescription snapshot  1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-mt5148-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        Friday Night Karaoke
      </h4>
      <p>
        It's Episode 50 of the Friday Night Karaoke Podcast, and the theme of the week was #FNKDedication.
      </p>
      <audio
        controls=""
        src="https://www.podtrac.com/pts/redirect.mp3/chrt.fm/track/35F11F/traffic.megaphone.fm/MOWM5486612603.mp3?updated=1684229072"
        style="width: 100%;"
      />
    </div>
  </div>
</div>
`;
